File: app/src/main/java/com/example/biblio/viewmodels/EbookDetailsViewModel.java
package com.example.biblio.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.biblio.helpers.LogHelper;

import org.jetbrains.annotations.NotNull;

import lrusso96.simplebiblio.core.Ebook;

public class EbookDetailsViewModel extends ViewModel {
    private final LogHelper logger = new LogHelper(getClass());
    private MutableLiveData<Ebook> ebook;

    public EbookDetailsViewModel() {
        if (ebook == null)
            ebook = new MutableLiveData<>();
    }

    public LiveData<Ebook> getEbook() {
        return ebook;
    }

    public void setEbook(@NotNull Ebook ebook) {
        logger.d(String.format("posting new ebook: %s", ebook.getTitle()));
        this.ebook.postValue(ebook);
    }
}

File: app/src/main/java/com/example/biblio/viewmodels/EbookDetailsViewModel.kt
package com.example.biblio.viewmodels

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.biblio.helpers.LogHelper
import lrusso96.simplebiblio.core.Ebook

class EbookDetailsViewModel : ViewModel() {
    private val logger = LogHelper(javaClass)
    val ebook: MutableLiveData<Ebook> = MutableLiveData()

    fun updateEbook(ebook: Ebook) {
        logger.d("posting new ebook: ${ebook.title}")
        this.ebook.postValue(ebook)
    }
}